builder/dockerfile/copy_windows.go

```diff
@@ -1,23 +1,104 @@
 package dockerfile
 
 import (
+	"bytes"
 	"errors"
+	"fmt"
+	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/Microsoft/go-winio"
 	"github.com/docker/docker/pkg/idtools"
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/docker/docker/pkg/system"
+	"golang.org/x/sys/windows"
 )
 
 var pathBlacklist = map[string]bool{
 	"c:\\":        true,
 	"c:\\windows": true,
 }
 
-func fixPermissions(source, destination string, rootIDs idtools.IDPair, overrideSkip bool) error {
-	// chown is not supported on Windows
+func init() {
+	reexec.Register("windows-fix-permissions", fixPermissionsReexec)
```

Comment:
None of this is `builder` specific, right? It's a general purpose "change permissions" ?

Could this be moved to `pkg/idtools` or maybe a new package?

Reply:
Technically it is specific to the builder, this is only needed so that ADD/COPY can change the permissions. In addition, pkd/idtools is vendored into dockerCLI and this code is not relevant there.

Reply:
Fair enough that `pkg/idtools` may not be the right place for it. 

Is it related to the builder? Isn't it just changing a files permissions? It just happens that it is used in build.

Reply:
fixPermissions is invoked for both Windows and Linux in the Builder (only). This code is using re-exec to avoid granting the entire process certain privileges. While we could put this in a separate package, it's not clear where the best place could be.

Since this is required for re-invoking the daemon as a new process, I'll check if somewhere there makes sense, or if the system package is another place to put them.

Reply:
`pkg/system` is a bit of a mix of random things, so I'd like to avoid that too.

It seems there isn't an existing package, so how about `pkg/windows` or `pkg/winpriv` or something like that?